src/main/java/chess/web/service/ChessService.java

```diff
@@ -1,106 +1,169 @@
+        Pieces pieces = Pieces.createInit();
         for (Piece piece : pieces.getPieces()) {
-            insertOrUpdatePiece(boardId, piece);
+            insertOrUpdatePiece(roomId, piece);
         }
-
-        Pieces savedPieces = Pieces.from(pieceDao.findAllByBoardId(boardId));
-        return Board.create(savedPieces, turn);
     }
 
-    private void insertOrUpdatePiece(Long boardId, Piece piece) {
+    private void insertOrUpdatePiece(Long roomId, Piece piece) {
         String position = piece.getPosition().name();
         String type = piece.getType();
         String team = piece.getTeam().value();
 
-        if (pieceDao.findByPositionAndBoardId(position, boardId).isPresent()) {
-            pieceDao.updatePieceByPositionAndBoardId(type, team, position, boardId);
+        if (pieceDao.findByPositionAndRoomId(position, roomId).isPresent()) {
+            pieceDao.updatePieceByPositionAndRoomId(type, team, position, roomId);
             return;
         }
-        pieceDao.save(piece, boardId);
+        pieceDao.save(piece, roomId);
     }
 
-    public ScoreDto getStatus(Long boardId) {
-        List<Piece> found = pieceDao.findAllByBoardId(boardId);
+    @Transactional(readOnly = true)
+    public ScoreDto getStatus(Long roomId) {
+        List<Piece> found = pieceDao.findAllByRoomId(roomId);
         Pieces pieces = Pieces.from(found);
 
         double blackScore = pieces.getTotalScore(Team.BLACK);
         double whiteScore = pieces.getTotalScore(Team.WHITE);
         return new ScoreDto(blackScore, whiteScore);
     }
+
+    @Transactional
+    public Long createRoom(String title, String password) {
+        Long id = roomDao.save(Turn.init().getTeam().value(), title, password);
+        pieceDao.save(Pieces.createInit().getPieces(), id);
+        return id;
+    }
+
+    @Transactional(readOnly = true)
+    public List<Room> getRooms() {
+        return roomDao.findAll();
+    }
+
+    @Transactional
+    public boolean removeRoom(Long roomId, String password) {
+        if (!canRemoveRoom(roomId, password)) {
+            return false;
+        }
+        pieceDao.deleteAllByRoomId(roomId);
+        roomDao.delete(roomId, password);
+        return true;
+    }
+
+    private boolean canRemoveRoom(Long roomId, String password) {
+        Optional<Room> optionalRoom = roomDao.findById(roomId);
+        if (optionalRoom.isEmpty()) {
+            return false;
+        }
+        Room room = optionalRoom.get();
+        if (isRunningChess(roomId, room) || !matchPassword(room, password)) {
+            return false;
+        }
+        return true;
+    }
+
+    private boolean isRunningChess(Long roomId, Room room) {
+        Turn turn = new Turn(Team.from(room.getTurn()));
+        Pieces pieces = Pieces.from(pieceDao.findAllByRoomId(roomId));
+        Board board = Board.create(pieces, turn);
+
+        boolean deadKing = board.isDeadKing();
+        return !deadKing;
```

Comment:
저라면 아래와 같이 수정할거 같아요.

```suggestion
        return board.isKingAlive();
```

제가 작성한 코드와 알린이 작성한 코드는 어떤 차이가 있을까요? 저는 두가지 정도가 생각나네요!

Reply:
제가 고민한 차이점입니다. 

1. isXXX()메서드를 사용하면서 영향을 주는 메서드들의 결을 맞추기 위해. 추가로 부정(!) 사용을 최소화하기 위해.
2. 저는 두 줄로 나누어 코드를 작성했는데 한줄로 줄일 수 있을 것 같다.

- is, has, can 등 boolean형을 반환하는 메서드는 가능하면 연관된 여러 메서드들의 결을 통일시키는 것이 좋다고 들었습니다.
- 예를 들면 연료가 있어야 자동차가 이동할 수 있다면 자동차가 이동할 수 있는지 확인하는 canMoveCar()라는 메서드에서 hasFuel()이라는 메서드를 쓰는것이 좋다는 의미입니다.
- 이런 통일성을 맞추기위 위한 기준은 보통 긍정형으로 메서드를 만드는 것이 좋다고 들었습니다.
- 예를들면 isNotKing()보다 isKing()이 좋다는 의미입니다.
- 이런 관점에서 `isRunningChess()`메서드에서는 `!isDeadKing()`과 `isKingAlive()`중 `isKingAlive()`를 사용하는 것이 더 좋아보입니다. 왕이 살아있어야 게임이 진행중이기 때문입니다.
- `isRunningChess()`와 `isDeadKing()`메서드의 특징이 다르기 때문에 반환을 하면서 !를 붙여줘야하고 코드의 이해도가 낮아질 것 같습니다.
- 심지어 프로덕션 코드에서 `if (!isRunningChess())`와 같이 사용되기 때문에 부정을 두번 하고 있어 더 이해도가 낮아질 것 같습니다.
- 이 문제를 해결하기 위해 `isKingAlive()`를 만들어 사용하거나 `isRunningChess()`를 `isFinishedChess()`로 수정하고 `isDeadKing()`를 반환하는 것도 방법이 될 것 같습니다.
- 저는 새로운 Board에 새로운 메서드를 만들지 않아도 되고 상태에 대한 확인이기 때문에 `isRunningChess()`를 `isFinishedChess()`로 수정하는 방식으로 수정해봤습니다.

Reply:
좋네요~! 제가 제안한 리뷰에 추가로 `isFinishedChess`로 수정하셨다는 내용을 보고 알린의 고민이 느껴졌습니다 😁
코드를 읽는 사람을 배려한 좋은 방향성이라 생각해요 👍 

적어주신 두가지 차이점도 제가 생각했던 내용과 동일해요. 2번을 조금 더 발전시키면 필요 없는 변수는 할당하지 말자 정도가 되겠네요.